### PR TITLE
Added Session Termination UseCase.

### PR DESCRIPTION
**Pull request by ujji8055**
## Related Issues
Terminate User's SSO sessions.

## Description
Terminate all the user's SSO sessions on receiving **Account Lockout Alert**.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.py [hunk 1]
```diff
@@ -559,6 +560,168 @@ def validate_tenant_sta(self):
...
 
+    # Get list of all the applications in the tenant.
+    def get_application_list_sta(self, limit=None):
+
+        uri = '/applications'
+        if limit:
+            limit = self.validate_limit_sta(limit)
+            if limit <= 100:
+                query_params = (
+                    ('pageIndex', 0),
```
Comment (daryakoval) [inline Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.py:571]: Is there an option to get to page number 3 using xsoar command?

File: Packs/SafeNet_Trusted_Access/IncidentTypes/incidenttype-SafeNet_Trusted_Access_Account_Lockout_Alert.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "name": "SafeNet Trusted Access Account Lockout Alert",
+    "onChangeRepAlg": 0,
+    "playbookId": "SafeNet Trusted Access - Terminate User SSO Sessions",
+    "readonly": false,
+    "reputationCalc": 0,
+    "system": false,
+    "version": -1,
+    "weeks": 0,
+    "weeksR": 0,
+    "fromVersion": "5.5.0"
```
Comment (daryakoval) [inline Packs/SafeNet_Trusted_Access/IncidentTypes/incidenttype-SafeNet_Trusted_Access_Account_Lockout_Alert.json:27]: If there is no specific reason for this incident type to has fromvesion 5.5.0, please change to 6.0.0.
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/SafeNet_Trusted_Access/Classifiers/classifier-SafeNet_Trusted_Access_–_Account_Lockout_Alert_Classifier.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{
+    "description": "Classifies SafeNet Trusted Access account lockout alerts.",
+    "feed": false,
+    "id": "accccf10-97da-425d-840f-beac0233efb1",
```
Comment (daryakoval) [inline Packs/SafeNet_Trusted_Access/Classifiers/classifier-SafeNet_Trusted_Access_–_Account_Lockout_Alert_Classifier.json:4]:
```suggestion
    "id": "SafeNet Trusted Access – Account Lockout Alert Classifier",
```

File: Packs/SafeNet_Trusted_Access/.pack-ignore [hunk 4]
```diff
@@ -1,5 +1,4 @@
-[file:SafeNet_Trusted_Access_–_Add_to_Unusual_Activity_Group.yml]
+[file:playbook-SafeNet_Trusted_Access_-_Add_to_Unusual_Activity_Group.yml]
+ignore=PB106
```
Comment (daryakoval) [inline Packs/SafeNet_Trusted_Access/.pack-ignore:2]: Why you are ignoring this error?

File: Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#### Integrations
+##### SafeNet Trusted Access
+- Added 5 commands.
```
[the remark this answers is not in the document]
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md:3]: Done. Updated as shown below.

image

File: Packs/SafeNet_Trusted_Access/README.md [hunk 6]
```diff
@@ -3,6 +3,7 @@ SafeNet Trusted Access prevents data breaches, helps with compliance with regula
 ## This pack enables you to:
 
 -	Receive and process SafeNet Trusted Access alerts that indicate security risks to end user accounts.
--	Apply security remediation actions on SafeNet Trusted Access.
+-	Apply security remediation actions on SafeNet Truste
+d Access.
```
[the remark this answers is not in the document]
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/README.md:7]: Done. Updated as shown below.

image

File: Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.yml [hunk 7]
```diff
@@ -711,6 +711,216 @@ script:
     - contextPath: STA.VALIDATE.TENANT
       description: Checks if you have permission to access the requested tenant.
       type: boolean
+  - arguments:
+    - description: The maximum number of results to return.
+      name: limit
```
[the remark this answers is not in the document]
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.yml:716]: No, API doesn't have an offset parameter. 

image

File: Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/README.md [hunk 8]
```diff
@@ -155,7 +155,7 @@ Get the profile information for a specific user.
 
 ### sta-create-user
 ***
-Create new user in the tenant
+Create a new user in the tenant.
```
[the remark this answers is not in the document]
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/README.md:158]: Done. Updated for every occurance.

image

File: Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.yml [hunk 9]
```diff
@@ -711,6 +711,216 @@ script:
...
+      type: string
+    - contextPath: STA.APPICATION.LASTMODIFIED
+      description: Last modified date and time of application.
+      type: string
+  - arguments:
+    - description: Username of the user.
+      name: userName
+      required: true
+    - description: The maximum number of results to return.
+      name: limit
```
[the remark this answers is not in the document]
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.yml:765]: By default (without limit parameter), it'll fetch information of the first page(index-0).
No default value for the limit. The maximum value is 10000. Validation is present in the code for the same.

File: Packs/SafeNet_Trusted_Access/Playbooks/playbook-SafeNet_Trusted_Access_-_Add_to_Unusual_Activity_Group.yml [hunk 10]
```diff
@@ -1,35 +1,35 @@
...
-  of time.
-id: SafeNet Trusted Access – Add to Unusual Activity Group
+  activity. SafeNet Trusted Access policies can be configured to take this into account
+  and provide stronger protection when handling access events from users who are members
+  of the group. The user is added to this group for a configurable period of time.
+id: SafeNet Trusted Access - Add to Unusual Activity Group
 inputs:
 - description: Username of the user.
-  key: userName
+  key: UserName
```
[the remark this answers is not in the document]
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/Playbooks/playbook-SafeNet_Trusted_Access_-_Add_to_Unusual_Activity_Group.yml:8]: This is suggested by the bussiness. As we have recently uploaded our previous versions and have less number of downloads. So it's ok.

File: Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.py (hunk 1 above)
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.py:571]: No such parameter/argument for the commands to change the page no.

File: Packs/SafeNet_Trusted_Access/IncidentTypes/incidenttype-SafeNet_Trusted_Access_Account_Lockout_Alert.json (hunk 2 above)
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/IncidentTypes/incidenttype-SafeNet_Trusted_Access_Account_Lockout_Alert.json:27]: Done.

image

File: Packs/SafeNet_Trusted_Access/Classifiers/classifier-SafeNet_Trusted_Access_–_Account_Lockout_Alert_Classifier.json (hunk 3 above)
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/Classifiers/classifier-SafeNet_Trusted_Access_–_Account_Lockout_Alert_Classifier.json:4]: Done. Updated ID to classifier name.

image

File: Packs/SafeNet_Trusted_Access/.pack-ignore (hunk 4 above)
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/.pack-ignore:2]: As discussed with Kobbi, this is affecting the usecase functionality. So as suggested, added this under the .pack-ignore file.
For our previous version pack, I am not able to see our playbook readme like https://xsoar.pan.dev/docs/reference/playbooks/abuse-inbox-management-detect--respond 
Please check and let us know the exact reason for the same.

Comment (kgal-pan) [reply Packs/SafeNet_Trusted_Access/.pack-ignore:2]: This error indicates that the Playbook is bound to a specific Integration instance. I informed Joshi that this is not best practice (that the Integration instance name is used as an input to a Playbook). 

Joshi informed me that they have a guide to that instructs users to specify the instance name.

Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/.pack-ignore:2]: Thank you Kobbi.

File: Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+#### Incident Types
+- **SafeNet Trusted Access Alert**
+- **SafeNet Trusted Access Account Lockout Alert**
+
+#### Playbooks
+##### New: SafeNet Trusted Access - Terminate User SSO Sessions
+- This playbook terminates user's SSO sessions so that upon the next login attempt following the unlocking of the account, authentication is required. (Available from Cortex XSOAR 6.0.0).
+
+##### Updated: SafeNet Trusted Access - Add to Unusual Activity Group
+- This playbook adds the user to a group that was created to identify unusual activity. SafeNet Trusted Access policies can be configured to take this into account and provide stronger protection when handling access events from users who are members of the group. The user is added to this group for a configurable period of time.
```
Comment (daryakoval) [inline Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md:27]:
```suggestion
- ***Breaking Change:*** Changed playbook inputs names. 
```

**Review by daryakoval — COMMENTED** [review]
Please run `demisto-sdk update-release-notes -i Packs/<Pack Name> -bc`, 
to add breaking change note, do not forget to change `breakingChangesNotes` field in the new created json file, as explained [here](https://xsoar.pan.dev/docs/documentation/release-notes#breaking-changes-version).

File: Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md (hunk 11 above)
Comment (ujji8055) [reply Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md]: Done. As discussed, updated release note and added the **.json** file.

**Comment by daryakoval** [comment]
Hi, you can convert timestamp to datastring in the human readable of sta-get-user-sessions command, using `timestamp_to_datestring` [function](https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L6443). 
***sta-get-user-application*** - check why this command not working.

**Comment by ujji8055** [comment]
@daryakoval 
The command is working as expected. As discussed within the team, there was some issue with our test environment Swagger Rest API. Please find the screenshot below.

image

**Comment by ujji8055** [comment]
@daryakoval 
Please put the pack on hold for now. I am incorporating Elia comments. I will let you know once it's good to merge.

**Comment by ujji8055** [comment]
@evisochek - I have created a common classifier and mapper for both the use-case. As discussed with Kobbi, added Deprecated into old classifier files and updated the incident type name for every occurrence.
Kindly review and suggest the needful.

**Review by daryakoval — APPROVED** [review]

**Pull request merged**